cellarium/ml/data/schema.py

```diff
@@ -60,9 +60,23 @@ def validate_anndata(self, adata: AnnData) -> None:
                         "of the reference anndata."
                     )
                 if not ref_value.dtypes.equals(value.dtypes):
+                    for col in ref_value.columns:
+                        if ref_value[col].dtype != value[col].dtype:
+                            if ref_value[col].dtype == "category":
+                                diff = set(ref_value[col].cat.categories).symmetric_difference(
+                                    set(value[col].cat.categories)
+                                )
+                                raise ValueError(
+                                    f".obs['{col}'].cat.categories for anndata passed in "
+                                    f"do not match those in the reference anndata. symmetric_differece: {diff}"
+                                )
+                            raise ValueError(
+                                f".obs['{col}'] dtype for anndata passed in ({value[col].dtype}) "
+                                f"does not match .obs['{col}'] dtype of the reference anndata ({ref_value[col].dtype})"
+                            )
                     raise ValueError(
```

Comment:
I think this part of the code becomes unreachable but not 100% sure (does line 64 do the same type of comparison as line 62?). But it is probably okay to keep it in case the comparisons are not identical.